### PR TITLE
KWS-2972: Post caption

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
https://jira.superawesome.tv/browse/KWS-2972

## Description
* Added CreateCaptionView

## Screenshots
caption

### CODE REVIEW

File: SAContentCreationTool/Classes/Views/CreationToolView.swift [hunk 1]
```diff
@@ -19,11 +20,13 @@ public class CreationToolView: UIView {
     
     // MARK: Constants
     
+    private let postImageSize = CGSize(width: 608.0, height: 608.0)
```
Comment (GuilhermeMota93) [inline SAContentCreationTool/Classes/Views/CreationToolView.swift]: Not entirely sure about this, so please feel free to ignore, but are we hardcoding sizes?

File: SAContentCreationTool/Classes/Views/CreationCaptionView.swift [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        textView.placeholderText = "Add a Caption".defaultLocalisation()
+        textView.accessibilityIdentifier = "caption_text_view"
+        textView.maximumMessageLength = 120
+        textView.isCharacterCounterHidden = false
+        textView.characterCounterTextAlignment = .left
+        textView.delegate = self
+        return textView
+    }
+    
+    private func buildPostButton() -> FlatButton {
```
Comment (Tbruni85) [inline SAContentCreationTool/Classes/Views/CreationCaptionView.swift]: why do you prefer this to:

`
private var flatButton: FlatButton = {

let buttonSkin = FlatButtonSkinModel(mainColor: Color.buttonColour1.value,
                                             selectedColor: Color.buttonColour1Pressed.value,
                                             iconImageName: nil,
                                             title: "POST".defaultLocalisation().uppercased(),
                                             buttonStyle: .fullColor)
        let button = FlatButton(WithSkin: buttonSkin)
        button.titleLabel?.font = Font.h1.value
        button.accessibilityIdentifier = "post_button"
        button.addTouchUpInsideAction {
            self.postPressed()
        }
  return button
}()
`

File: SAContentCreationTool/Classes/Views/CreationCaptionView.swift [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    private func backPressed() {
+        delegate?.backButtonPressed(withCreateCaptionView: self)
+    }
+}
+
+// MARK: TaggableTextViewDelegate
+
+extension CreateCaptionView: TaggableTextViewDelegate {
+    public func textDidChange(withTaggableTextView taggableTextView: TaggableTextView) {
+    }
```
Comment (Tbruni85) [inline SAContentCreationTool/Classes/Views/CreationCaptionView.swift]: public func textDidChange(withTaggableTextView taggableTextView: TaggableTextView) {}

File: SAContentCreationTool/Classes/Views/CreationToolView.swift [hunk 4]
```diff
@@ -221,3 +251,16 @@ extension CreationToolView: CreationToolViewModelDelegate {
...
+// MARK: CreateCaptionViewDelegate
+
+extension CreationToolView: CreateCaptionViewDelegate {
+    public func postButtonPressed(withCaption caption: String?, image: UIImage, createCaptionView: CreateCaptionView) {
+        createCaptionView.removeFromSuperview()
+        postImageOnFeed(withImage: image, caption: caption)
+    }
+    
+    public func backButtonPressed(withCreateCaptionView createCaptionView: CreateCaptionView) {
+
```
Comment (Tbruni85) [inline SAContentCreationTool/Classes/Views/CreationToolView.swift]: public func backButtonPressed(withCreateCaptionView createCaptionView: CreateCaptionView) {}

File: SAContentCreationTool/Classes/Views/CreationToolView.swift (hunk 1 above)
Comment (me2k) [reply SAContentCreationTool/Classes/Views/CreationToolView.swift]: This is the post size, so it should be hard coded unless we can get it from a configuration

**Pull request merged**